src/resharper-unity/annotations/UnityEngine.xml

```diff
@@ -0,0 +1,14 @@
+<assembly name="UnityEngine">
+  <member name="P:UnityEngine.Component.transform">
+    <attribute ctor="M:JetBrains.Annotations.NotNullAttribute.#ctor" />
+  </member>
+  <member name="P:UnityEngine.Component.gameObject">
+    <attribute ctor="M:JetBrains.Annotations.NotNullAttribute.#ctor" />
+  </member>
+  <member name="P:UnityEngine.GameObject.transform">
+    <attribute ctor="M:JetBrains.Annotations.NotNullAttribute.#ctor" />
```

Comment:
Same here, XML docs, and [website](http://docs.unity3d.com/ScriptReference/GameObject-transform.html) state `transform` can be `null`.